### ISSUE TITLE
ping fails when IPv6 is disabled

### ISSUE THREAD

**Issue by lonaowna** [iputils/iputils#32]
When `ping` is ran without the `-4` and `-6` flags, the `create_socket` function is ran for both protocols. This gives the following error on machines which have disabled IPv6 in the kernel:
````
ping: socket: Address family not supported by protocol (raw socket required by specified options).
````

The program should not fail when an IPv6 socket cannot be opened, if an IPv4 socket is already available.

[Arch Linux bug report](https://bugs.archlinux.org/task/46297)

**Comment by Techwolf** [comment]
When I first install my current OS, I disable ipv6 for performance reasons, no ISP support, and to avoid bugs when network has no ipv6 connection.

**Comment by okias** [comment]
Can you verify, that from this pull request [1], patch "ping: handle single protocol systems" solve problem?

[1] https://github.com/iputils/iputils/pull/28

**Comment by lonaowna** [comment]
I've tested it, and #28 completely fixes the problem!

**Comment by okias** [comment]
Fixed in git :)

**Comment by xftroxgpx** [comment]
**tl;dr:** to have ping work when ran as a non-root user, this must be so:
```bash
$ getcap `which ping`
/usr/bin/ping = cap_net_raw+ep
```

------

Hey just wanted to leave a note here in case anyone stumbles upon this(from searches like I did) somewhat related case where ping from root works but from local user doesn't.
I did a `rsync -a /oldroot/ /newroot` but I should've done `rsync -X -a /oldroot/ /newroot` (-X is --xattrs                preserve extended attributes), on ArchLinux. At this time only two pacman packages seem to make use of `setcap`(in the `.INSTALL` file of the /var/cache/pacman/pkg/iputils-20161105.1f2bb12-2-x86_64.pkg.tar.xz package) and they are `iputils` and `inetutils`; the former contains `ping`.
The first rsync didn't copy the `CAP_NET_RAW`
>              * Use RAW and PACKET sockets;
>              * bind to any address for transparent proxying.
 and that's why the error:
`ping: socket: Address family not supported by protocol` when ran as local user(but not when ran as root)
So just reinstalling the package(s) fixed it: `$ sudo pacman -S iputils inetutils`

This is the relevant part in `.INSTALL` of `iputils` package, if anyone wants to do this manually or not on ArchLinux:

```bash
post_install() {
  setcap cap_net_raw=ep usr/bin/ping  2>/dev/null || chmod +s usr/bin/ping
}
```
If the `setcap` above succeeded then `ping` won't show up as suid, but will show red colored:
```bash
$ ls -la `which ping`
-rwxr-xr-x 1 root root 61168 11.02.2017 13:06 /usr/bin/ping*
```
(imagine `/usr/bin/ping` is red colored)


Cheers!
